go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -468,14 +468,38 @@ func (d *DoltSession) NewPendingCommit(ctx *sql.Context, dbName string, roots do
 		return nil, err
 	}
 
+	headCommit := sessionState.headCommit
+	headHash, _ := headCommit.HashOf()
+
 	var mergeParentCommits []*doltdb.Commit
 	if sessionState.WorkingSet.MergeActive() {
 		mergeParentCommits = []*doltdb.Commit{sessionState.WorkingSet.MergeState().Commit()}
+	} else if props.Amend {
+		numParentsHeadForAmend := headCommit.NumParents()
+		for i := 0; i < numParentsHeadForAmend; i++ {
+			parentCommit, err := headCommit.GetParent(ctx, i)
```

Comment:
We should add handling for when this `err` is not nil. Swallowed errors are a pain to debug!

Reply:
Good point, adding handling code.